core/src/main/java/org/kiwix/kiwixmobile/core/webserver/ZimHostActivity.kt

```diff
@@ -0,0 +1,290 @@
+        Toast.LENGTH_SHORT
+      ).show()
+    }
+  }
+
+  private fun stopServer() {
+    startService(createHotspotIntent(ACTION_STOP_SERVER))
+  }
+
+  private fun select(bookOnDisk: BooksOnDiskListItem.BookOnDisk) {
+    val booksList: ArrayList<BooksOnDiskListItem> = ArrayList()
+    for (item in booksAdapter.items) {
+      if (item == bookOnDisk) {
+        item.isSelected = !item.isSelected
+      }
+      booksList.add(item)
+    }
+    booksAdapter.items = booksList
+    saveHostedBooks(booksList)
+  }
+
+  override fun onStart() {
+    super.onStart()
+    bindService()
+  }
+
+  override fun onStop() {
+    super.onStop()
+    unbindService()
+  }
+
+  private fun bindService() {
+    bindService(
+      Intent(this, HotspotService::class.java), serviceConnection,
+      Context.BIND_AUTO_CREATE
+    )
+  }
+
+  private fun unbindService() {
+    if (hotspotService != null) {
+      unbindService(serviceConnection)
+      hotspotService!!.registerCallBack(null)
+    }
+  }
+
+  override fun onResume() {
+    super.onResume()
+    presenter.loadBooks(sharedPreferenceUtil.hostedBooks)
+    if (ServerUtils.isServerStarted) {
+      ip = ServerUtils.getSocketAddress()
+      layoutServerStarted()
+    }
+  }
+
+  private fun saveHostedBooks(booksList: ArrayList<BooksOnDiskListItem>) {
+    sharedPreferenceUtil.hostedBooks =
+      booksList.filter { book -> book.isSelected && book is BookOnDisk && book.book.title != null }
+        .map { book -> (book as BookOnDisk).book.title }.toSet()
+  }
+
+  private fun layoutServerStarted() {
+    serverTextView.text = getString(R.string.server_started_message, ip)
+    startServerButton.text = getString(R.string.stop_server_label)
+    startServerButton.setBackgroundColor(resources.getColor(R.color.stopServer))
+    bookDelegate.selectionMode = SelectionMode.NORMAL
+    booksAdapter.notifyDataSetChanged()
+  }
+
+  private fun layoutServerStopped() {
+    serverTextView.text = getString(R.string.server_textview_default_message)
+    startServerButton.text = getString(R.string.start_server_label)
+    startServerButton.setBackgroundColor(resources.getColor(R.color.greenTick))
+    bookDelegate.selectionMode = SelectionMode.MULTI
+    booksAdapter.notifyDataSetChanged()
+  }
+
+  override fun onDestroy() {
+    super.onDestroy()
+    presenter.detachView()
+  }
+
+  private fun setUpToolbar() {
+    val toolbar = findViewById<Toolbar>(R.id.toolbar)
+    setSupportActionBar(toolbar)
+    supportActionBar!!.title = getString(R.string.menu_host_books)
+    supportActionBar!!.setHomeButtonEnabled(true)
+    supportActionBar!!.setDisplayHomeAsUpEnabled(true)
+
+    toolbar.setNavigationOnClickListener { onBackPressed() }
+  }
+
+  // Advice user to turn on hotspot manually for API<26
+  private fun startHotspotManuallyDialog() {
+
+    alertDialogShower.show(KiwixDialog.StartHotspotManually(),
+      {
+        launchTetheringSettingsScreen()
+        Unit
```

Comment:
Unit can now be omitted, it was a workaround for the java code

Reply:
https://github.com/kiwix/kiwix-android/pull/1569/commits/73dea16d9ec7a5811d8ba96bea7136ad5003af37